src/OSPSuite.Presentation/Presenters/Importer/ColumnMappingPresenter.cs

```diff
@@ -145,44 +144,45 @@ public void SetDescriptionForRow(ColumnMappingDTO model)
          _setDescriptionForRow(model, values != null && values.All(v => v != model.ExcelColumn));
       }
 
-      public void UpdateMetaDataForModel()
+      public void UpdateMetaDataForModel(MetaDataFormatParameter mappingSource)
       {
-         if (_currentModel == null) return;
-
-         var metaData = _currentModel.Source as MetaDataFormatParameter;
-         if (metaData == null) return;
+         if (mappingSource == null)
+            return;
 
-         _mappings.First(m => m.MappingName == metaData.MetaDataId).ExcelColumn = _metaDataParameterEditorPresenter.Input;
-         metaData.ColumnName = _metaDataParameterEditorPresenter.Input;
-         metaData.IsColumn = false;
+         _mappings.First(m => m.MappingName == mappingSource.MetaDataId).ExcelColumn = _metaDataParameterEditorPresenter.Input;
```

Comment:
First will throw an exception if not found...Or it cannot happen in this scenario?

Reply:
Well in that case it would mean that we changed the unit for a mapping that doesn't exist - which would really be a software bug. I would let the exception be raised i this quite improbable case.